pkg/cluster/k8sres.go

```diff
@@ -949,6 +954,20 @@ func (c *Cluster) generateStatefulSet(spec *acidv1.PostgresSpec) (*appsv1.Statef
 	return statefulSet, nil
 }
 
+func (c *Cluster) generatePodAnnotations(spec *acidv1.PostgresSpec) map[string]string {
+	annotations := make(map[string]string)
+	for k, v := range c.OpConfig.CustomPodAnnotations {
+		annotations[k] = v
+	}
+	if spec == nil {
+		return annotations
+	}
+	for k, v := range spec.PodAnnotations {
+		annotations[k] = v
+	}
```

Comment:
This function should return `nil` when no annotations are set. Otherwise this leads to rolling updates during each sync as the operator finds a mismatch between the current emtpy annotations map `map[string]string(nil)` and the new `map[string]string{}`. Change it to something like:
```
	if spec != nil {
		for k, v := range spec.PodAnnotations {
			annotations[k] = v
		}
	}
	if len(annotations) == 0 {
		return nil
	}
```